### DISCUSSION TITLE
Replace hdbscan in bertopic with k-means and evaluate whether bertopic can be evaluated using topic consistency metrics？

### DISCUSSION THREAD

**Discussion by susanhhhhhh** [MaartenGr/BERTopic#2127] [Q&A]
Replace hdbscan in bertopic with k-means and evaluate whether bertopic can be evaluated using topic consistency metrics

**Comment by MaartenGr** [comment]
You can indeed replace HDBSCAN with k-Means (or any other clustering method) and apply topic-based metrics using tools like Gensim or Octis.

**Reply by susanhhhhhh** [reply]
Hello, the k value in k-means needs to be manually adjusted. Can I use the theme consistency metric to determine the optimal number of theme k values

**Reply by MaartenGr** [reply]
What is the "theme consistency metric"? I would advise looking into metrics related to both the clustering itself (so metrics related specifically to the cluster algorithm k-means) and metrics that relate to the topic modeling itself (like topic diversity, coherence, etc.). All in all, it depends on your specific use case which metrics are relevant.

**Reply by susanhhhhhh** [reply]
Okay, thank you for your answer!
